src/libcollections/str.rs

```diff
@@ -273,7 +273,7 @@ impl<'a> Iterator for Recompositions<'a> {
         loop {
             match self.state {
                 Composing => {
-                    for ch in self.iter {
+                    while let Some(ch) = self.iter.next() {
```

Comment:
Actually, if it helps in basically all cases, I wonder if we could spec it like this...